universal-application-tool-0.0.1/app/controllers/LoginController.java

```diff
@@ -0,0 +1,53 @@
+package controllers;
+
+import java.util.Optional;
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.inject.Named;
+import org.pac4j.core.context.session.SessionStore;
+import org.pac4j.core.exception.http.RedirectionAction;
+import org.pac4j.core.http.adapter.HttpActionAdapter;
+import org.pac4j.oidc.client.OidcClient;
+import org.pac4j.play.PlayWebContext;
+import org.pac4j.play.http.PlayHttpActionAdapter;
+import play.mvc.Controller;
+import play.mvc.Http;
+import play.mvc.Result;
+
+/**
+ * These routes can be hit even if you are logged in already, which is what allows the merge logic -
+ * normally you need to be logged out in order to be redirected to a login page.
+ */
+public class LoginController extends Controller {
+  @Inject
```

Comment:
Question (blocking): why not use c'tor injection? Unless there's a reason to deviate I'd prefer we use a consistent pattern

Reply:
Eh, thought it was uglier since there are so many annotations, but I put it to constructor injection.